server-phoenix/lib/helios_web/controllers/web_hooks/slack_controller.ex

```diff
@@ -1,12 +1,76 @@
 defmodule HeliosWeb.WebHooks.SlackController do
   use HeliosWeb, :controller
-  alias Helios.{Repo, Event}
+  require Logger
+  alias Helios.{Repo, Event, SlackChannelNames}
+
+  defp insert_channel_id(channel_id) do
+    unless SlackChannelNames
+           |> SlackChannelNames.with_id(channel_id)
+           |> Repo.exists?() do
+      try do
+        body = %{
+          "channel" => channel_id
+        }
+
+        request_body = URI.encode_query(body)
+
+        headers = [
+          {"Accept", "application/json"},
+          {"Authorization",
+           "Bearer #{Application.get_env(:helios, HeliosWeb.Endpoint)[:slack_bearer_token]}"},
+          {"Content-Type", "application/x-www-form-urlencoded"}
+        ]
+
+        channel_response =
+          HTTPoison.post!("https://slack.com/api/conversations.info", request_body, headers,
+            follow_redirect: true
+          )
+
+        json = Jason.decode!(channel_response.body)
+        name = json["channel"]["name"]
+
+        Repo.insert!(%SlackChannelNames{
+          channel_id: channel_id,
+          channel_name: name
+        })
+      rescue
+        e -> IO.inspect(e)
+      end
+    end
+  end
 
   def handle(conn, params) do
     if params["type"] == "url_verification" do
       send_resp(conn, 200, params["challenge"])
     else
       event = params["event"]
+      img = params["event"]["files"]
+
+      if img do
+        url = Enum.at(img, 0)["url_private_download"]
+
+        headers = [
+          Authorization:
+            "Bearer #{Application.get_env(:helios, HeliosWeb.Endpoint)[:slack_bearer_token]}"
+        ]
+
+        Task.Supervisor.start_child(
+          SlackImageDownloader,
+          fn ->
+            try do
+              response = HTTPoison.post!(url, [], headers, follow_redirect: true)
+            rescue
+              e -> IO.inspect(e)
```

Comment:
Again, not sure what we're rescuing here. If this isn't caught it'll propagate up to the `Task.Supervisor` which will log the crash for us.